plugins/Compressor/CompressorControlDialog.cpp

```diff
@@ -0,0 +1,753 @@
+	{
+		QRectF knee2Rect = QRect(m_lastKneePoint, 0, kneePoint - m_lastKneePoint, m_kneeWindowSizeY);
+		p.drawPixmap(knee2Rect, m_kneePixmap, knee2Rect);
+	}
+	else
+	{
+		p.setCompositionMode(QPainter::CompositionMode_Source);
+		p.fillRect(kneePoint, 0, m_lastKneePoint, m_kneeWindowSizeY, QColor("transparent"));
+		p.setCompositionMode(QPainter::CompositionMode_SourceOver);
+	}
+	m_lastKneePoint = kneePoint;
+
+	p.end();
+
+	if (m_controls->m_effect->m_redrawThreshold)
+	{
+		p.begin(&m_miscPixmap);
+
+		p.setCompositionMode(QPainter::CompositionMode_Source);
+		p.fillRect(0, 0, m_windowSizeX, m_windowSizeY, QColor("transparent"));
+		p.setCompositionMode(QPainter::CompositionMode_SourceOver);
+
+		p.setRenderHint(QPainter::Antialiasing, true);
+
+		// Draw threshold lines
+		p.setPen(QPen(m_threshColor, 2, Qt::DotLine));
+		p.drawLine(0, threshYPoint, m_windowSizeX, threshYPoint);
+		p.drawLine(threshXPoint, 0, threshXPoint, m_kneeWindowSizeY);
+
+		p.end();
+
+		m_controls->m_effect->m_redrawThreshold = false;
+	}
+
+	m_lastPoint = yPoint;
+	m_lastGainPoint = yGainPoint;
+
+	update();
+}
+
+
+
+void CompressorControlDialog::paintEvent(QPaintEvent *event)
+{
+	if (isVisible())
+	{
+		QPainter p(this);
+
+		p.setCompositionMode(QPainter::CompositionMode_Source);
+		p.fillRect(0, 0, m_windowSizeX, m_windowSizeY, QColor("transparent"));
+		p.setCompositionMode(QPainter::CompositionMode_SourceOver);
+
+		p.drawPixmap(0, 0, m_graphPixmap);
+		p.drawPixmap(0, 0, m_visPixmap);
+		p.setOpacity(0.25);
+		p.drawPixmap(0, 0, m_kneePixmap);
+		p.setOpacity(1);
+		if (m_controls->m_effect->isEnabled() && m_controls->m_effect->isRunning())
+		{
+			p.drawPixmap(0, 0, m_kneePixmap2);
+		}
+		p.drawPixmap(0, 0, m_miscPixmap);
+
+		p.end();
+	}
+}
+
+
+inline int CompressorControlDialog::dbfsToYPoint(float inDbfs)
+{
+	return (-((inDbfs + m_dbRange) / m_dbRange) + 1) * m_windowSizeY;
+}
+
+inline int CompressorControlDialog::dbfsToXPoint(float inDbfs)
+{
+	return m_kneeWindowSizeY - dbfsToYPoint(inDbfs);
+}
+
+
+void CompressorControlDialog::resizeEvent(QResizeEvent *event)
+{
+	resetCompressorView();
+}
+
+
+void CompressorControlDialog::wheelEvent(QWheelEvent * event)
+{
+	const float temp = m_dbRange;
+	m_dbRange = round(qBound(3.f, m_dbRange - event->delta() / 20.f, 96.f) / 3.f) * 3.f;
+
+	// Only reset view if the scolling had an effect
+	if (m_dbRange != temp)
+	{
+		resetGraph();
+		m_controls->m_effect->m_redrawKnee = true;
+		m_controls->m_effect->m_redrawThreshold = true;
+	}
+}
+
+
+void CompressorControlDialog::resetGraph()
+{
+	QPainter p;
+
+	p.begin(&m_graphPixmap);
+
+	p.setRenderHint(QPainter::Antialiasing, false);
+
+	p.setCompositionMode(QPainter::CompositionMode_Source);
+	p.fillRect(0, 0, m_windowSizeX, m_windowSizeY, QColor("transparent"));
+	p.setCompositionMode(QPainter::CompositionMode_SourceOver);
+
+	p.setPen(QPen(m_textColor, 1));
+	p.setFont(QFont("Arial", qMax(int(m_windowSizeY / 1080.f * 24), 12)));
```

Comment:
Where does this font's `pointSize` calculation come from and will it work on all systems?

Reply:
This is for changing the font size when the window is resized.  The font size calculation is entirely arbitrary, I just changed the formula until it looked good.  It just increases the font size by 1 for every 45 pixels in the screen's height, while not allowing the font size to go below 12.
It... should work on all systems.  In what circumstances could it potentially not work?

Reply:
> It... should work on all systems. In what circumstances could it potentially not work?

We never know. Just wanted to clarify!